### ISSUE TITLE
Ubuntu Error

### ISSUE THREAD

**Issue by dirkkeaton** [beanumber/retro#3]
Good morning,

After running "tar xvzf chadwick-0.7.2.tar.gz" I received the following message

tar (child): chadwick-0.7.2.tar.gz: Cannot open: No such file or directory
tar (child): Error is not recoverable: exiting now
tar: Child returned status 2
tar: Error is not recoverable: exiting now

Thanks so much.

Best,
Dirk

**Comment by beanumber** [comment]
Do you have that file downloaded in that directory? What does `ls -l` say?

**Comment by dirkkeaton** [comment]
Yep, it worked after I changed the directory.  Was about to send your a retraction, but thank for the werich response.  I'm noticing one more potential issue but i think I can navigate.

**Comment by dirkkeaton** [comment]
Okay, so I ran into another issue, and I'm a little stumped.  I downloaded mysql, and after I ran
_> db <- src_mysql_cnf("retrosheet")_
I got the following:
_Warning message:
In src_mysql_cnf("retrosheet") :
  No MySQL config file found...trying anyway..._
When I ran:
_ls -1 /etc/mysql_
it showed the existence of a config file.

Any advice?

**Comment by beanumber** [comment]
You need to create [a MySQL option file](https://dev.mysql.com/doc/refman/5.7/en/option-files.html) if you want to use `src_mysql_cnf()`. It is looking for a file at `~/.my.cnf` with the relevant parameters for the MySQL database you want to connect to. 

Alternatively, you can use the function `src_mysql()` instead of `src_mysql_cnf()`, and just specify the parameters as arguments to that function.

**Comment by dirkkeaton** [comment]
Here's the current state of my thinking (I've been doing more reading and I think I know a bit more--to explain, I'm seasoned in R, never used SQL before, returning to Ubuntu after about a decade)

I attempted to run:

`src_mysql(retro, host = NULL, port = 0L, user = "root", password = "")`

And, I got some messages indicating that I need to download the RSql library.  When I try to install that, it seems that I have to download  MariaDB (not the library, the program).
I have opened a new tab and attempted to download MariaDB through the terminal, but I'm also receiving errors.

So, I took a look at option files, and from my really brief reading (I admit the article you shared seems a little over my head)takes the form:

```
[name of database]
user=
password=
host=IP
Port=port number
database=name of database

```
Given that I'm working on my own computer which isn't a server (though I have installed mysql server, perhaps wrongly) the file should look something like:

```
[retro]
user=
password=
host=127.0.0.1
Port=3306
database=retro
```
But, it seems odd that I would be able to do that without having installed all of the other programs and libraries I mentioned earlier.  So, I admit to being at a loss.  One thing I suspect I have done wrong is that I've never bothered to set up a username, password, or really, do anything with the mysql server that I downloaded since I downloaded it, essentially, for the purpose of getting your code to work.

So, I wrote a book, and suspect that I need to re-learn ubuntu and learn more about sql, but thanks for your help.

**Comment by beanumber** [comment]
Right. You need to install a local MySQL Server, and then figure out how to log in, and put those credentials in the MySQL options file. 

Or, if you have access to another MySQL Server, you can use those credentials (I'm guessing that you're not in this boat). 

Or, you can try to use a SQLite database instead of MySQL. SQLite is a little easier to use in that you don't have to set up a server ahead of time. 

Bottom line is that you need to have a SQL database somewhere to put the data -- that's sort of the whole point of this package. That said, setting that database up is not something this package will help you with. You may find [this book chapter helpful](https://beanumber.github.io/mdsr2e/ch-db-setup.html#mysql).

So I'm going to close this issue, but feel free to keep posting here and I can try to keep helping.

**Comment by dirkkeaton** [comment]
Thanks so much for all of your help.  The chapter from your book made the whole process of using the option file very easy, and I've come further than ever before.  I do have a few questions:
After downloading the databases, I recieved the error:
```
Error: Can't rename columns that don't exist.
✖ The column `Var1` doesn't exist.
Run `rlang::last_error()` to see where the error occurred.

```
But, when I looked through my files, the aforementioned zip files seem to be located where they ought to be, so I assume the error is trivial.

After running the second command from your page, I received the following error:
```

> dbRunScript(retro$con, 
+             system.file("sql", "optimize.mysql", package = "retro"))
Initializing DB using SQL script optimize.mysql
Error: near "USE": syntax error

```
Should I be troubled by either of these?  Again, thanks so much.

Best,
Dirk

**Comment by beanumber** [comment]
What command did you run that generated the `Var1` error?

**Comment by dirkkeaton** [comment]
It happened after I ran

`retro <- etl("retro", db = db, dir = "~/dumps/retro/")`

**Comment by dirkkeaton** [comment]
I hate to follow up so quickly, but there is some related stangeness.  I tried installing RMySQL and I keep getting this error message.  

```
* installing *source* package ‘RMySQL’ ...
** package ‘RMySQL’ successfully unpacked and MD5 sums checked
** using staged installation
Using PKG_CFLAGS=
Using PKG_LIBS=-lmysqlclient
-----------------------------[ ANTICONF ]-----------------------------
Configure could not find suitable mysql/mariadb client library. Try installing:
 * deb: libmariadbclient-dev | libmariadb-client-lgpl-dev (Debian, Ubuntu)
 * rpm: mariadb-connector-c-devel | mariadb-devel | mysql-devel (Fedora, CentOS, RHEL)
 * csw: mysql56_dev (Solaris)
 * brew: mariadb-connector-c (OSX)
If you already have a mysql client library installed, verify that either
mariadb_config or mysql_config is on your PATH. If these are unavailable
you can also set INCLUDE_DIR and LIB_DIR manually via:
R CMD INSTALL --configure-vars='INCLUDE_DIR=... LIB_DIR=...'
--------------------------[ ERROR MESSAGE ]----------------------------
<stdin>:1:10: fatal error: mysql.h: No such file or directory
compilation terminated.
-----------------------------------------------------------------------
ERROR: configuration failed for package ‘RMySQL’
* removing ‘/usr/local/lib/R/site-library/RMySQL’

The downloaded source packages are in
	‘/tmp/Rtmp9DCVS2/downloaded_packages’
Warning message:
In install.packages("RMySQL") :
  installation of package ‘RMySQL’ had non-zero exit status

```
Because the package seemed to be presupposing mariadb, I installed it and then added my r-user to it (it, now, is reading my option file and opening when I type mysql in the terminal).  However, it is still giving me the error when I try to install the package.  

It also mentions a path.  It seems to be the case that mysql is on my path (if it wasn't), then I wouldn't be able to type mysql from terminal and have it work if it wasn't on my path, from what I understand.  That being said, I can't find files named mariadb_config

While it seems to suggest that I should run:
`
R CMD INSTALL --configure-vars='INCLUDE_DIR=... LIB_DIR=...'`

from the terminal, I'm trying to understand what variables to use for the directories or if I'm, again, looking down a blind alley.

Thanks so much for all of your help.

**Comment by beanumber** [comment]
You may have mariadb installed, but you need the development headers files to compile RMySQL. So at the command line:

```
sudo apt-get install libmariadbclient-dev
```

Then try to reinstall `RMySQL` from within R.

**Comment by dirkkeaton** [comment]
In a hilarious turn, I accidentally installed mariadb 10.5 and the dev headers are only compatible with 10.3.  I'm trying ODBC instead (since it seems to be a popular alternative to RMySQL.
